pkg/services/provisioning/dashboards/dashboard.go

```diff
@@ -3,44 +3,79 @@ package dashboards
 import (
 	"context"
 	"fmt"
-
 	"github.com/grafana/grafana/pkg/log"
+	"github.com/pkg/errors"
 )
 
-type DashboardProvisioner struct {
-	cfgReader *configReader
-	log       log.Logger
+type DashboardProvisioner interface {
```

Comment:
Creating interface here for testing.

Reply:
This is the interface I moved to the api package which moves it closer to the caller instead of implementer. Maybe not the best example. But I like the idea of exposing structs globally (using inject) and having local interfaces. I think we could reduce the dependency graph quite a lot if we did it in more places. For the scope if this PR we dont need to change this. I just wanted to comment on it since we talked briefly about it at the standup.

Reply:
I wonder if it is possible you will get something else from registry than you want that way? I mean could you have an interface and 2 items in the registry would match that interface but would be totaly different? Like 3 services for provisioning with Provision() func that is required by something but all 3 implementations (dashboards/datasources/notifs) match that? I if this is an issue it could happen even now, but with more streamlined interfaces that just define requirements from the code that needs the dependency, having a non discriminating interface like that would be more likely.